outputparser/defined_test.go

```diff
@@ -0,0 +1,137 @@
+package outputparser
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestDefinedNonStruct(t *testing.T) {
+	t.Parallel()
+	var arr []struct{}
+
+	if _, err := NewDefined(arr); err == nil {
+		t.Error("schema source can only be a struct")
+	}
+}
+
+func TestDefinedInvalidStructs(t *testing.T) {
+	t.Parallel()
+	tests := map[string]any{
+		"empty struct":       struct{}{},
+		"no fields with tag": struct{ Field string }{},
+	}
+	for name, input := range tests {
+		if _, err := NewDefined(input); err == nil {
+			t.Errorf("expected error for %s", name)
+		}
+	}
+}
+
+func TestDefined(t *testing.T) {
+	t.Parallel()
+	type Shape struct {
+		Name     string `json:"shapeName" describe:"shape name"`
+		NumSides int    `json:"numSides" describe:"number of sides"`
+	}
+
+	tests := map[string]struct {
```

Comment:
please include more test cases (separate PR ok) that include wantErr: true cases